### ISSUE TITLE
Nice template (missing install and configure) instruction

### ISSUE THREAD

**Issue by SBerg1980** [avenger11/Apple-HomePlay#2]
Looks very good but i cannot get it to work, maybe you can write a installation and configure instruction.

**Comment by avenger11** [comment]
Yeah, this is the plan. But I think I will have to rethink my folder/ yaml structure. It would be nice if there is a way to have an installation package per "widget".

**Comment by tubbers** [comment]
agree. would love to install this

**Comment by avenger11** [comment]
@tubbers  @SBerg1980  I revamp the folder/file structure and separate the code so only one file need to be modified for integration with other dashboard.

Can you go through the draft documentation and let me know what's missing ? https://github.com/avenger11/Apple-HomePlay/wiki

I started by the integration of the weather card. By following the documentation/instruction it should be possible to integrate it inside your own dashboard.

**Comment by mikekenyon99** [comment]
Hi - I agree I would like to install this but I'm guessing there isn't an integration for this.  Instead you are taking a standard dashboard and putting all the element together yourself?

Thanks

Mike

**Comment by mikekenyon99** [comment]
Had a good play with this tonight and I can't get any of it to work.  For some reason I have put homeplay.yaml in the directory like your folder structure and HA doesn't pick it up when I go to Dashboards in the UI.  Sample from my configuration.yaml:

```
lovelace:
  mode: yaml
  dashboards:
    lovelace-homeplay:
      mode: yaml
      filename: homeplay.yaml
      title: HomePlay
      icon: phu:homekit
      show_in_sidebar: true
```


 If I create your code from the front end it doesn't understand the lines:

```
button_card_templates: !include_dir_merge_named button_card_templates

tap_action: !include popup/temperature.yaml
```
basically the inline !includes. 

It's really hard to understand what I need to put where and how to config HA to accept it.  Is the logic that a user build up the dashboard themselves or do they use your code? 

I want to support and use this but at the moment, the best I can do is copy the ideas and then try and implement myself. 

Thanks 

Mike

**Comment by avenger11** [comment]
Hi @mikekenyon99 ,

First Thanks for trying ! 

Start by the [Wiki](https://github.com/avenger11/Apple-HomePlay/wiki)
I'm currently recode / separating everything from scratch

It will be easier to share and understand how it works. 

This part is still under development so not everything is coded back yet.
![image](https://github.com/avenger11/Apple-HomePlay/assets/37946892/996e1fcd-d83a-49f1-b087-80771287add2)

**Comment by tobyv275** [comment]
Sounds great. I have tested the weather "widget" and the calendar "widget". Both work great. I have problems with the person part. See screenshot. 
![Screenshot 2023-10-02 214954](https://github.com/avenger11/Apple-HomePlay/assets/68601642/f729a91d-29e2-48b3-ae86-9810ec0b9b21)

I think this is a great design setup, maybe the best I've seen so far. Would like to use it. How can I support you Seb ;) ?

**Comment by avenger11** [comment]
@tobyv275 So I guess my explanation and the new structured way work better ! Nice !
As I'm splitting the code I didn't finish the person card yet. :sweat_smile:
But I finalised the new structure (grid in a grid to keep it simple to edit)

![image](https://github.com/avenger11/Apple-HomePlay/assets/37946892/d0cffa43-7436-43d3-945e-d4f87b479acb)

I move the appbar on the side, that make it easier to have square card and a bit bigger card overall.

This is a mook-up of the final look i'm going for:
![image](https://github.com/avenger11/Apple-HomePlay/assets/37946892/27e75f88-0967-4fee-acc4-5f5f8b2c2d38)

Glad you want to help !, here are some area that need hands :

- [ ] fixe the person card 
- [ ]  port one the of the other card from Homeplay.yaml (washing machine, recycling etc..) 
- [ ] help with documentation to help other users 

let me know what you think !

**Comment by tobyv275** [comment]
looks great!!!
I am looking forward to test this ;) 
I think my coding skills aren't as good as yours :D So I would like to help you with the documentation and donate you a coffee 👍

**Comment by avenger11** [comment]
@tobyv275 Any help is always appreciated ! :)

**Comment by CommittotheIndian** [comment]
So I'm trying to give this dashboard a shot as well and been struggling with some of the setup. I've been chipping away before having to reach out but I'm starting to hit some walls with configuration. It probably doesn't help that some of the code is French but google translate is helping.

- I have the weather integrated with my entities but I can't pull the expected image into the dashboard. What template triggers the image? I'm guessing my weather integration doesn't feed to your template as expected.

- I have the calendar configured per the instructions (best I can tell). I have my calendar integrated with HA so it shouldn't be an issue with my credentials. Have you considered using HA's local calendar?

- I got your floor plan image added (for testing) but I'm getting code overlaid. Haven't even started working on entities. This overlay is related to the calendar card above. When I remove it from the HomeOS.yaml, I remove that overlay. Without the map (floor plan) I don't get the error even with the calendar card enabled so it's making it difficult to decipher the error.>>> **Part of this screenshot was fixed by adding a sensor in config for time_date. This exposed sensor.date.**

- If I add the Apps Bar or Top Bar definitions all of the proportions and screen gets squished. I will admit, I have yet to check this on a tablet since I'm still in the coding phase on a desktop monitor. 

I am more than willing to work on testing for you but may need some guidance at times.

![image](https://github.com/avenger11/Apple-HomePlay/assets/60204668/c3d82e28-f44c-43d5-b417-1ddbd1dd1ea5)

**Comment by avenger11** [comment]
Hey folks, 

Sorry for the no radio. 
I was having to many issues with my HA installation so I change a lot. 
I put everything into container  (HA, Node red, zigbee2mqtt, eqmx, sql database) and I use Zigbee over Ethernet (no more usb plug).
Now, when I update HA every week, my entire system is not down for 10-15mins.


Anyway, I created a new project HomeOS based of the learning/sharing of this version. 
It's still in early days but I separate even more the code from the user defined entity.

@CommittotheIndian i will see how to have options for the weather card to use a OOTB weather entity

Cheers,

**Comment by CommittotheIndian** [comment]
@avenger11 

Thanks for the update. I've been chipping away trying to get my setup to work.

For the weather card, I am using a template to change the states provided in openweatherapi into the numeric code of you provide. https://github.com/avenger11/Apple-HomePlay/issues/6#issuecomment-1772542613

I have my calendar working but only 1 calendar pulls in. Any reason it wouldn't list multiple calendars?

I understand the left app bar and the top bar. I haven't figured out the bottom and right areas.

**Comment by avenger11** [comment]
The code for the calendar currently is for one only. If you want to add a second calendar you wil have to go into the calendar_card.yaml and add a second calendar there:

    custom_fields:
      agenda:
        card:
          type: custom:atomic-calendar-revive
          entities:
            - entity: calendar.famille
            - entity: calendar.NameOfYourSecondCalendar

For the bottom and right areas I didn't recode yet in HomeOS dashboard

**Comment by CommittotheIndian** [comment]
@avenger11 

What size did you export the photos for the map? It looked like the dimensions were 1000x887 but clearly that isn't right.

My floor plan is a bit bigger. I might rework it with less detail (currently to scale). Any advise on building my map would be appreciated.

All of this thread, is building toward your needed documentation.

![image](https://github.com/avenger11/Apple-HomePlay/assets/60204668/5a4c7782-cbe1-472c-8cde-e27be73313c7)

**Comment by CommittotheIndian** [comment]
@avenger11 
I was just curious if you have any updates on your development? You've mentioned a few times of changes you've made the code or different directions with a module but as I keep digging in on building my dashboard, I worry it will be changed completely with an update or you will just drop the project all together.

**Comment by avenger11** [comment]
> @avenger11 
> 
> I was just curious if you have any updates on your development? You've mentioned a few times of changes you've made the code or different directions with a module but as I keep digging in on building my dashboard, I worry it will be changed completely with an update or you will just drop the project all together.

I created a new repo that is way more clear.
You can check the code to see what change. I try as much as possible to separate the user definition from the code. It will be easier to choose where you want the card to be displayed as well within the grid.
https://github.com/avenger11/HomeOS
